src/scopes.js

```diff
@@ -56,3 +60,35 @@ exports.scopeMatch = function(scopePatterns, scopesets) {
     });
   });
 };
+
+/**
+ * Finds scope intersections between two scope sets.
+ */
+exports.scopeIntersection = (scopeset1, scopeset2) => {
+  validateScopePatterns(scopeset2);
+  validateScopePatterns(scopeset1);
+
+  const junctions = [];
+  const scope1 = scopeset2.filter(s => s);
+  const scope2 = scopeset1.filter(s => s);
+
+  scope2.forEach((s2) => {
+    scope1.forEach((s1) => {
+      const star = '*';
+      const s1IndexOfStar = s1.indexOf(star);
+      const s2IndexOfStar = s2.indexOf(star);
+
+      if (s1 === s2) {
+        junctions.push(s1);
+      } else if (s1.slice(-1) === star && s1.slice(0, s1IndexOfStar) === s2.slice(0, s1IndexOfStar)) {
```

Comment:
`s1IndexOfStar` should probably be replaced with `s1.length - 1` right?

Reply:
Yikes!